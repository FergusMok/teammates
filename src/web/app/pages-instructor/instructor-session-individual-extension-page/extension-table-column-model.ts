export interface StudentExtensionTableColumnModel {
    sectionName: string;
    teamName: string;
    name: string;
    email: string;
    extensionDeadline: number;
    hasExtension: boolean;
    selected: boolean;
  }

  export interface InstructorExtensionTableColumnModel {
    institute?: string;
    name: string;
    email: string;
    extensionDeadline: number;
    hasExtension: boolean;
    selected: boolean;
  }
