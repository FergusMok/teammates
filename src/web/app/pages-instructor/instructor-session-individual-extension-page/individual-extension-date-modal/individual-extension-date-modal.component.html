<div class="modal-header alert-primary">
  <h5 class="modal-title">Select the new deadline for extension</h5>
  <button type="button" class="close" (click)="activeModal.dismiss()">
    <i class="fas fa-times"></i>
  </button>
</div>
<div class="modal-body">
  Extending the feedback session for <b>{{ numStudents }}</b> student(s) and <b>{{ numInstructors }}</b> instructor(s):
</div>

<div class="container ml-1">

  <div class="row pl-3">
    <label class="mr-2"><input type="radio" name="optradio" [value]="RadioOptions.EXTEND_BY" [(ngModel)]="radioOption">
      Extend deadline by
    </label>
    <label><input type="radio" name="optradio" [(ngModel)]="radioOption" [value]="RadioOptions.EXTEND_TO">
      Extend deadline to
    </label>
  </div>

  <div *ngIf="isRadioExtendBy()">
    <small>Extend the feedback session by:</small>
    <div id="submission-start-date">
        <div class="input-group w-75">
            <select class="form-control" [(ngModel)]="extendByDeadlineKey">
              <option *ngFor="let deadlineOption of extendByDeadlineOptions | keyvalue: sortMapByOriginalOrder" [value]="deadlineOption.key">
                {{ deadlineOption.key }}
                <div *ngIf="deadlineOption.key !== 'Customize'">
                  ({{ addTimeAndFormat(deadlineOption.value, 0) }})
                </div>
              </option>
            </select>
        </div>
    </div>
  </div>

  <div *ngIf="isRadioExtendTo()">
    <small>The new deadline of the feedback session will be:</small>
    <div id="submission-start-date">
      <div class="input-group">
        <tm-datepicker [date]="datePicker"
          (dateChangeCallback)="onChangeDateTime($event,'date')"
          [minDate]="getDateFormat(feedbackSessionEndingTime)">
        </tm-datepicker>
        <div class="col-md-5">
          <tm-timepicker id="submission-end-time"
            [time]="timePicker"
            (timeChange)="onChangeDateTime($event,'time')">
          </tm-timepicker>
        </div>
      </div>
    </div>
  </div>

  <div *ngIf="isCustomize()">
      <form class="container">
        <div class="row">
          <div class="col-sm pl-0">
            <small>Hours</small>
            <input class="form-control" id="form-hours" name="form-hours" type="text" placeholder="Hours" [(ngModel)]="extendByDatePicker.hours">
          </div>
          <div class="col-sm">
            <small>Days</small>
            <input class="form-control" id="form-days" name="form-days" type="text" placeholder="Days" [(ngModel)]="extendByDatePicker.days">
          </div>
        </div>
      </form>
    <br>
    <p class="mb-0">
      The new extended deadline will be: {{ addTimeAndFormat(extendByDatePicker.hours, extendByDatePicker.days) }}
    </p>
  </div>

  <br>
</div>

<div class="modal-footer">
  <button type="button" class="btn btn-light" (click)="activeModal.dismiss()">No, cancel the Operation</button>
  <button type="button" class="btn btn-success modal-btn-ok" (click)="onConfirm()" [disabled]="!isValidForm()">
    Save Changes</button>
</div>
