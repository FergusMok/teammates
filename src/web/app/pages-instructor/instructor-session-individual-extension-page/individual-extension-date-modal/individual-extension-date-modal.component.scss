.hours {
  padding-left: 0;
}
