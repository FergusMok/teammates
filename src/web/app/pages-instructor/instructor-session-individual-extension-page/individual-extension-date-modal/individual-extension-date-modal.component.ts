import { Component, EventEmitter, Input, Output } from '@angular/core';
import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
import moment from 'moment-timezone';
import { SimpleModalService } from '../../../../services/simple-modal.service';
import { TimezoneService } from '../../../../services/timezone.service';
import { DateFormat } from '../../../components/datepicker/datepicker.component';
import { SimpleModalType } from '../../../components/simple-modal/simple-modal-type';
import { FormatDateDetailPipe } from '../../../components/teammates-common/format-date-detail.pipe';
import { TimeFormat } from '../../../components/timepicker/timepicker.component';

export enum RadioOptions {
  EXTEND_TO = 1,
  EXTEND_BY = 2,
}

@Component({
  selector: 'tm-individual-extension-date-modal',
  templateUrl: './individual-extension-date-modal.component.html',
  styleUrls: ['./individual-extension-date-modal.component.scss'],
})
export class IndividualExtensionDateModalComponent {
  @Input()
  numStudents: number = 0;

  @Input()
  numInstructors: number = 0;

  @Input()
  feedbackSessionEndingTime: number = 0;

  @Input()
  feedbackSessionTimeZone: string = '';

  @Output() onConfirmCallBack: EventEmitter<number> = new EventEmitter();

  constructor(
    public activeModal: NgbActiveModal,
    private timeZoneService: TimezoneService,
    private simpleModalService: SimpleModalService,
  ) {}

  RadioOptions: typeof RadioOptions = RadioOptions;
  radioOption: RadioOptions = RadioOptions.EXTEND_BY;
  extendByDeadlineKey: String = '';
  extendByDeadlineOptions: Map<String, Number> = new Map([
    ['12 hours', 0.5],
    ['1 day', 24],
    ['3 days', 72],
    ['1 week', 168],
    ['Customize', 0],
  ]);

  ONE_MINUTE_IN_MILLISECONDS = 60 * 1000;
  ONE_HOUR_IN_MILLISECONDS = 60 * this.ONE_MINUTE_IN_MILLISECONDS;
  ONE_DAY_IN_MILLISECONDS = 24 * this.ONE_HOUR_IN_MILLISECONDS;
  datePicker: DateFormat = { year: 0, month: 0, day: 0 };
  timePicker: TimeFormat = { hour: 23, minute: 59 };
  extendByDatePicker = { hours: 0, days: 0 };
  dateDetailPipe = new FormatDateDetailPipe(this.timeZoneService);

  onConfirm(): void {
    if (this.getExtensionTimestamp() < Date.now()) {

      const extensionTimeString = this.dateDetailPipe.transform(
        this.getExtensionTimestamp(),
        this.feedbackSessionTimeZone,
      );
      const currentTimeString = this.dateDetailPipe.transform(
        Date.now(),
        this.feedbackSessionTimeZone,
      );
      this.simpleModalService
        .openConfirmationModal(
          'Are you sure you wish to set the new deadline to before the current time?',
          SimpleModalType.WARNING,
          '<b>Any users affected will have their sessions closed immediately.</b>'
            + ` The current time now is ${currentTimeString} and you are extending to`
            + ` ${extensionTimeString}. Do you wish to proceed?`,
        )
        .result.then(() => this.onConfirmCallBack.emit(this.getExtensionTimestamp()));
    } else {
      this.onConfirmCallBack.emit(this.getExtensionTimestamp());
    }
  }

  onChangeDateTime(data: DateFormat | TimeFormat, field: string): void {
    if (field === 'date') {
      this.datePicker = data as DateFormat;
    } else if (field === 'time') {
      this.timePicker = data as TimeFormat;
    }
  }

  getDateFormat(timestamp: number): DateFormat {
    let momentInstance: moment.Moment = moment(timestamp);
    if (momentInstance.hour() === 0 && momentInstance.minute() === 0) {
      momentInstance = momentInstance.subtract(1, 'minute');
    }
    return {
      year: momentInstance.year(),
      month: momentInstance.month() + 1, // moment return 0-11 for month
      day: momentInstance.date(),
    };
  }

  getExtensionTimestamp(): number {
    if (this.isRadioExtendBy()) {
      if (this.isCustomize()) {
        return this.addTime(this.feedbackSessionEndingTime, this.extendByDatePicker.hours,
          this.extendByDatePicker.days);
      }
      if (this.extendByDeadlineOptions.has(this.extendByDeadlineKey)) {
        return this.addTime(this.feedbackSessionEndingTime,
          this.extendByDeadlineOptions.get(this.extendByDeadlineKey)!.valueOf(), 0,
        );
      }
    }
    if (this.isRadioExtendTo()) {
      return this.timeZoneService.resolveLocalDateTime(this.datePicker, this.timePicker, this.feedbackSessionTimeZone,
        true);
    }
    return this.feedbackSessionEndingTime;
  }

  addTimeAndFormat(hours: number, days: number): string {
    const time = this.addTime(this.feedbackSessionEndingTime, hours, days);
    return this.dateDetailPipe.transform(time, this.feedbackSessionTimeZone);
  }

  private addTime(timestamp: number, hours: number, days: number): number {
    return timestamp + hours * this.ONE_HOUR_IN_MILLISECONDS + days * this.ONE_DAY_IN_MILLISECONDS;
  }

  isValidForm(): boolean {
    return this.getExtensionTimestamp() > this.feedbackSessionEndingTime;
  }

  isRadioExtendBy(): boolean {
    return this.radioOption === RadioOptions.EXTEND_BY;
  }

  isRadioExtendTo(): boolean {
    return this.radioOption === RadioOptions.EXTEND_TO;
  }

  isCustomize(): boolean {
    return this.isRadioExtendBy() && this.extendByDeadlineKey === 'Customize';
  }

  sortMapByOriginalOrder = (): number => {
    return 0;
  };
}
