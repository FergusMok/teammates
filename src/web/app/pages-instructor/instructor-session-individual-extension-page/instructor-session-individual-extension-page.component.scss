.background-color-medium-gray {
  background-color: #E5E5E6;
}

.remind-all {
  width: 250px;
}

.clickable {
  cursor: pointer;
}

.margin-right-5px {
  margin-right: 5px;
}

.card-plain {
  background-color: #EAEFF5;
  background-image: none;
  border: 0;
  box-shadow: 0 0 0 0;
}

.close-header {
  padding-top: .75rem;
  padding-right: 1.25rem;
  margin-bottom: 0;
}

.font-bold {
  font-weight: bold;
}

.border-left-gray {
  border-left: 1px solid #DDD;
}

.margin-top-20px {
  margin-top: 20px;
}

.session-form-buttons > button {
  margin-left: var(--btn-margin);
}

.align-row {
  display: flex;
  flex-direction: row;
  gap: 20px;
}
