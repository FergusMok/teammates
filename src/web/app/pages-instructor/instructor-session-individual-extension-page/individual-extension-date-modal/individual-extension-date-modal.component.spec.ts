import { HttpClientTestingModule } from '@angular/common/http/testing';
import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
import SpyInstance = jest.SpyInstance;
import { SimpleModalService } from '../../../../services/simple-modal.service';
import { TimezoneService } from '../../../../services/timezone.service';
import { createMockNgbModalRef } from '../../../../test-helpers/mock-ngb-modal-ref';
import {
  FeedbackSession,
  FeedbackSessionPublishStatus,
  FeedbackSessionSubmissionStatus,
  ResponseVisibleSetting,
  SessionVisibleSetting,
} from '../../../../types/api-output';
import { InstructorSessionIndividualExtensionPageModule } from '../instructor-session-individual-extension-page.module';
import { IndividualExtensionDateModalComponent, RadioOptions } from './individual-extension-date-modal.component';

describe('IndividualExtensionDateModalComponent', () => {
  const testFeedbackSession: FeedbackSession = {
    courseId: 'testId1',
    timeZone: 'Asia/Singapore',
    feedbackSessionName: 'Test Session',
    instructions: 'Instructions',
    submissionStartTimestamp: 1000000000000,
    submissionEndTimestamp: 1500000000000,
    gracePeriod: 0,
    sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
    responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
    submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
    publishStatus: FeedbackSessionPublishStatus.PUBLISHED,
    isClosingEmailEnabled: true,
    isPublishedEmailEnabled: true,
    createdAtTimestamp: 0,
    studentDeadlines: {},
    instructorDeadlines: {},
  };

  const testTimeString = '5 Apr 2000 2:00:00';

  let component: IndividualExtensionDateModalComponent;
  let fixture: ComponentFixture<IndividualExtensionDateModalComponent>;
  let simpleModalService: SimpleModalService;
  let timeZoneService: TimezoneService;

  beforeEach(
    waitForAsync(() => {
      TestBed.configureTestingModule({
        imports: [HttpClientTestingModule, InstructorSessionIndividualExtensionPageModule],
        providers: [NgbActiveModal],
      }).compileComponents();
    }),
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(IndividualExtensionDateModalComponent);
    component = fixture.componentInstance;
    simpleModalService = TestBed.inject(SimpleModalService);
    timeZoneService = TestBed.inject(TimezoneService);
    jest.spyOn(timeZoneService, 'formatToString').mockReturnValue(testTimeString);
    component.numStudents = 10;
    component.numInstructors = 20;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with the extended students', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with the extend by radio option', () => {
    component.extendByDeadlineKey = '12 hours';
    component.radioOption = RadioOptions.EXTEND_TO;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with the extend by radio option with customize', () => {
    component.extendByDeadlineKey = 'Customize';
    component.radioOption = RadioOptions.EXTEND_TO;
    component.extendByDatePicker = { hours: 20, days: 100 };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with the extend to radio option with timepicker', () => {
    component.radioOption = RadioOptions.EXTEND_BY;
    component.datePicker = { year: 2022, month: 10, day: 10 };
    component.timePicker = { hour: 10, minute: 30 };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with the error modal', () => {
    component.feedbackSessionTimeZone = testFeedbackSession.timeZone;
    component.radioOption = RadioOptions.EXTEND_BY;
    component.datePicker = { year: 2020, month: 10, day: 10 };
    component.DATETIME_FORMAT = 'd MMM YYYY h:mm:ss';
    const modalSpy: SpyInstance = jest
      .spyOn(simpleModalService, 'openConfirmationModal')
      .mockReturnValue(createMockNgbModalRef());

    component.onConfirm();

    expect(modalSpy).toHaveBeenCalledTimes(1);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
