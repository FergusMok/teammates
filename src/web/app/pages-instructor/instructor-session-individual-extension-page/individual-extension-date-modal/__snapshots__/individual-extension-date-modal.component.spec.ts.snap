// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndividualExtensionDateModalComponent should snap with the error modal 1`] = `
<tm-individual-extension-date-modal
  DATETIME_FORMAT={[Function String]}
  ONE_DAY_IN_MILLISECONDS={[Function Number]}
  ONE_HOUR_IN_MILLISECONDS={[Function Number]}
  ONE_MINUTE_IN_MILLISECONDS={[Function Number]}
  RadioOptions={[Function Object]}
  activeModal={[Function NgbActiveModal]}
  datePicker={[Function Object]}
  extendByDatePicker={[Function Object]}
  extendByDeadlineKey=""
  extendByDeadlineOptions={[Function Map]}
  feedbackSessionEndingTime="0"
  feedbackSessionTimeZone={[Function String]}
  numInstructors={[Function Number]}
  numStudents={[Function Number]}
  onConfirmCallBack={[Function EventEmitter_]}
  radioOption={[Function Number]}
  simpleModalService={[Function SimpleModalService]}
  sortMapByOriginalOrder={[Function Function]}
  timePicker={[Function Object]}
  timeZoneService={[Function TimezoneService]}
>
  <div
    class="modal-header alert-primary"
  >
    <h5
      class="modal-title"
    >
      Select the new deadline for extension
    </h5>
    <button
      class="close"
      type="button"
    >
      <i
        class="fas fa-times"
      />
    </button>
  </div><div
    class="modal-body"
  >
     Extending the feedback session for 
    <b>
      10
    </b>
     student(s) and 
    <b>
      20
    </b>
     instructor(s):
  </div><div
    class="container ml-1"
  >
    <div
      class="row pl-3"
    >
      <label
        class="mr-2"
      >
        <input
          class="ng-untouched ng-pristine ng-valid"
          name="optradio"
          type="radio"
        />
         Extend deadline by 
      </label>
      <label>
        <input
          class="ng-untouched ng-pristine ng-valid"
          name="optradio"
          type="radio"
        />
         Extend deadline to 
      </label>
    </div>
    <div>
      <small>
        Extend the feedback session by:
      </small>
      <div
        id="submission-start-date"
      >
        <div
          class="input-group w-75"
        >
          <select
            class="form-control ng-untouched ng-pristine ng-valid"
          >
            <option
              value="12 hours"
            >
               12 hours 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="1 day"
            >
               1 day 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="3 days"
            >
               3 days 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="1 week"
            >
               1 week 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="Customize"
            >
               Customize 
            </option>
          </select>
        </div>
      </div>
    </div>
    <br />
  </div><div
    class="modal-footer"
  >
    <button
      class="btn btn-light"
      type="button"
    >
      No, cancel the Operation
    </button>
    <button
      class="btn btn-success modal-btn-ok"
      disabled=""
      type="button"
    >
       Save Changes
    </button>
  </div>
</tm-individual-extension-date-modal>
`;

exports[`IndividualExtensionDateModalComponent should snap with the extend by radio option 1`] = `
<tm-individual-extension-date-modal
  DATETIME_FORMAT={[Function String]}
  ONE_DAY_IN_MILLISECONDS={[Function Number]}
  ONE_HOUR_IN_MILLISECONDS={[Function Number]}
  ONE_MINUTE_IN_MILLISECONDS={[Function Number]}
  RadioOptions={[Function Object]}
  activeModal={[Function NgbActiveModal]}
  datePicker={[Function Object]}
  extendByDatePicker={[Function Object]}
  extendByDeadlineKey={[Function String]}
  extendByDeadlineOptions={[Function Map]}
  feedbackSessionEndingTime="0"
  feedbackSessionTimeZone=""
  numInstructors={[Function Number]}
  numStudents={[Function Number]}
  onConfirmCallBack={[Function EventEmitter_]}
  radioOption={[Function Number]}
  simpleModalService={[Function SimpleModalService]}
  sortMapByOriginalOrder={[Function Function]}
  timePicker={[Function Object]}
  timeZoneService={[Function TimezoneService]}
>
  <div
    class="modal-header alert-primary"
  >
    <h5
      class="modal-title"
    >
      Select the new deadline for extension
    </h5>
    <button
      class="close"
      type="button"
    >
      <i
        class="fas fa-times"
      />
    </button>
  </div><div
    class="modal-body"
  >
     Extending the feedback session for 
    <b>
      10
    </b>
     student(s) and 
    <b>
      20
    </b>
     instructor(s):
  </div><div
    class="container ml-1"
  >
    <div
      class="row pl-3"
    >
      <label
        class="mr-2"
      >
        <input
          class="ng-untouched ng-pristine ng-valid"
          name="optradio"
          type="radio"
        />
         Extend deadline by 
      </label>
      <label>
        <input
          class="ng-untouched ng-pristine ng-valid"
          name="optradio"
          type="radio"
        />
         Extend deadline to 
      </label>
    </div>
    <div>
      <small>
        The new deadline of the feedback session will be:
      </small>
      <div
        id="submission-start-date"
      >
        <div
          class="input-group"
        >
          <tm-datepicker>
            <div
              class="input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                ngbdatepicker=""
                type="text"
              />
              <div
                class="input-group-append"
              >
                <button
                  class="btn btn-light"
                  type="button"
                >
                  <i
                    class="fas fa-calendar-alt"
                  />
                </button>
              </div>
            </div>
          </tm-datepicker>
          <div
            class="col-md-5"
          >
            <tm-timepicker
              id="submission-end-time"
            >
              <select
                class="form-control ng-untouched ng-pristine ng-valid"
              >
                <option
                  value="1: Object"
                >
                   01:00H 
                </option>
                <option
                  value="2: Object"
                >
                   02:00H 
                </option>
                <option
                  value="3: Object"
                >
                   03:00H 
                </option>
                <option
                  value="4: Object"
                >
                   04:00H 
                </option>
                <option
                  value="5: Object"
                >
                   05:00H 
                </option>
                <option
                  value="6: Object"
                >
                   06:00H 
                </option>
                <option
                  value="7: Object"
                >
                   07:00H 
                </option>
                <option
                  value="8: Object"
                >
                   08:00H 
                </option>
                <option
                  value="9: Object"
                >
                   09:00H 
                </option>
                <option
                  value="10: Object"
                >
                   10:00H 
                </option>
                <option
                  value="11: Object"
                >
                   11:00H 
                </option>
                <option
                  value="12: Object"
                >
                   12:00H 
                </option>
                <option
                  value="13: Object"
                >
                   13:00H 
                </option>
                <option
                  value="14: Object"
                >
                   14:00H 
                </option>
                <option
                  value="15: Object"
                >
                   15:00H 
                </option>
                <option
                  value="16: Object"
                >
                   16:00H 
                </option>
                <option
                  value="17: Object"
                >
                   17:00H 
                </option>
                <option
                  value="18: Object"
                >
                   18:00H 
                </option>
                <option
                  value="19: Object"
                >
                   19:00H 
                </option>
                <option
                  value="20: Object"
                >
                   20:00H 
                </option>
                <option
                  value="21: Object"
                >
                   21:00H 
                </option>
                <option
                  value="22: Object"
                >
                   22:00H 
                </option>
                <option
                  value="23: Object"
                >
                   23:00H 
                </option>
                <option
                  value="0: Object"
                >
                  23:59H
                </option>
              </select>
            </tm-timepicker>
          </div>
        </div>
      </div>
    </div>
    <br />
  </div><div
    class="modal-footer"
  >
    <button
      class="btn btn-light"
      type="button"
    >
      No, cancel the Operation
    </button>
    <button
      class="btn btn-success modal-btn-ok"
      disabled=""
      type="button"
    >
       Save Changes
    </button>
  </div>
</tm-individual-extension-date-modal>
`;

exports[`IndividualExtensionDateModalComponent should snap with the extend by radio option with customize 1`] = `
<tm-individual-extension-date-modal
  DATETIME_FORMAT={[Function String]}
  ONE_DAY_IN_MILLISECONDS={[Function Number]}
  ONE_HOUR_IN_MILLISECONDS={[Function Number]}
  ONE_MINUTE_IN_MILLISECONDS={[Function Number]}
  RadioOptions={[Function Object]}
  activeModal={[Function NgbActiveModal]}
  datePicker={[Function Object]}
  extendByDatePicker={[Function Object]}
  extendByDeadlineKey={[Function String]}
  extendByDeadlineOptions={[Function Map]}
  feedbackSessionEndingTime="0"
  feedbackSessionTimeZone=""
  numInstructors={[Function Number]}
  numStudents={[Function Number]}
  onConfirmCallBack={[Function EventEmitter_]}
  radioOption={[Function Number]}
  simpleModalService={[Function SimpleModalService]}
  sortMapByOriginalOrder={[Function Function]}
  timePicker={[Function Object]}
  timeZoneService={[Function TimezoneService]}
>
  <div
    class="modal-header alert-primary"
  >
    <h5
      class="modal-title"
    >
      Select the new deadline for extension
    </h5>
    <button
      class="close"
      type="button"
    >
      <i
        class="fas fa-times"
      />
    </button>
  </div><div
    class="modal-body"
  >
     Extending the feedback session for 
    <b>
      10
    </b>
     student(s) and 
    <b>
      20
    </b>
     instructor(s):
  </div><div
    class="container ml-1"
  >
    <div
      class="row pl-3"
    >
      <label
        class="mr-2"
      >
        <input
          class="ng-untouched ng-pristine ng-valid"
          name="optradio"
          type="radio"
        />
         Extend deadline by 
      </label>
      <label>
        <input
          class="ng-untouched ng-pristine ng-valid"
          name="optradio"
          type="radio"
        />
         Extend deadline to 
      </label>
    </div>
    <div>
      <small>
        The new deadline of the feedback session will be:
      </small>
      <div
        id="submission-start-date"
      >
        <div
          class="input-group"
        >
          <tm-datepicker>
            <div
              class="input-group"
            >
              <input
                class="form-control ng-untouched ng-pristine ng-valid"
                ngbdatepicker=""
                type="text"
              />
              <div
                class="input-group-append"
              >
                <button
                  class="btn btn-light"
                  type="button"
                >
                  <i
                    class="fas fa-calendar-alt"
                  />
                </button>
              </div>
            </div>
          </tm-datepicker>
          <div
            class="col-md-5"
          >
            <tm-timepicker
              id="submission-end-time"
            >
              <select
                class="form-control ng-untouched ng-pristine ng-valid"
              >
                <option
                  value="1: Object"
                >
                   01:00H 
                </option>
                <option
                  value="2: Object"
                >
                   02:00H 
                </option>
                <option
                  value="3: Object"
                >
                   03:00H 
                </option>
                <option
                  value="4: Object"
                >
                   04:00H 
                </option>
                <option
                  value="5: Object"
                >
                   05:00H 
                </option>
                <option
                  value="6: Object"
                >
                   06:00H 
                </option>
                <option
                  value="7: Object"
                >
                   07:00H 
                </option>
                <option
                  value="8: Object"
                >
                   08:00H 
                </option>
                <option
                  value="9: Object"
                >
                   09:00H 
                </option>
                <option
                  value="10: Object"
                >
                   10:00H 
                </option>
                <option
                  value="11: Object"
                >
                   11:00H 
                </option>
                <option
                  value="12: Object"
                >
                   12:00H 
                </option>
                <option
                  value="13: Object"
                >
                   13:00H 
                </option>
                <option
                  value="14: Object"
                >
                   14:00H 
                </option>
                <option
                  value="15: Object"
                >
                   15:00H 
                </option>
                <option
                  value="16: Object"
                >
                   16:00H 
                </option>
                <option
                  value="17: Object"
                >
                   17:00H 
                </option>
                <option
                  value="18: Object"
                >
                   18:00H 
                </option>
                <option
                  value="19: Object"
                >
                   19:00H 
                </option>
                <option
                  value="20: Object"
                >
                   20:00H 
                </option>
                <option
                  value="21: Object"
                >
                   21:00H 
                </option>
                <option
                  value="22: Object"
                >
                   22:00H 
                </option>
                <option
                  value="23: Object"
                >
                   23:00H 
                </option>
                <option
                  value="0: Object"
                >
                  23:59H
                </option>
              </select>
            </tm-timepicker>
          </div>
        </div>
      </div>
    </div>
    <br />
  </div><div
    class="modal-footer"
  >
    <button
      class="btn btn-light"
      type="button"
    >
      No, cancel the Operation
    </button>
    <button
      class="btn btn-success modal-btn-ok"
      disabled=""
      type="button"
    >
       Save Changes
    </button>
  </div>
</tm-individual-extension-date-modal>
`;

exports[`IndividualExtensionDateModalComponent should snap with the extend to radio option with timepicker 1`] = `
<tm-individual-extension-date-modal
  DATETIME_FORMAT={[Function String]}
  ONE_DAY_IN_MILLISECONDS={[Function Number]}
  ONE_HOUR_IN_MILLISECONDS={[Function Number]}
  ONE_MINUTE_IN_MILLISECONDS={[Function Number]}
  RadioOptions={[Function Object]}
  activeModal={[Function NgbActiveModal]}
  datePicker={[Function Object]}
  extendByDatePicker={[Function Object]}
  extendByDeadlineKey=""
  extendByDeadlineOptions={[Function Map]}
  feedbackSessionEndingTime="0"
  feedbackSessionTimeZone=""
  numInstructors={[Function Number]}
  numStudents={[Function Number]}
  onConfirmCallBack={[Function EventEmitter_]}
  radioOption={[Function Number]}
  simpleModalService={[Function SimpleModalService]}
  sortMapByOriginalOrder={[Function Function]}
  timePicker={[Function Object]}
  timeZoneService={[Function TimezoneService]}
>
  <div
    class="modal-header alert-primary"
  >
    <h5
      class="modal-title"
    >
      Select the new deadline for extension
    </h5>
    <button
      class="close"
      type="button"
    >
      <i
        class="fas fa-times"
      />
    </button>
  </div><div
    class="modal-body"
  >
     Extending the feedback session for 
    <b>
      10
    </b>
     student(s) and 
    <b>
      20
    </b>
     instructor(s):
  </div><div
    class="container ml-1"
  >
    <div
      class="row pl-3"
    >
      <label
        class="mr-2"
      >
        <input
          class="ng-untouched ng-pristine ng-valid"
          name="optradio"
          type="radio"
        />
         Extend deadline by 
      </label>
      <label>
        <input
          class="ng-untouched ng-pristine ng-valid"
          name="optradio"
          type="radio"
        />
         Extend deadline to 
      </label>
    </div>
    <div>
      <small>
        Extend the feedback session by:
      </small>
      <div
        id="submission-start-date"
      >
        <div
          class="input-group w-75"
        >
          <select
            class="form-control ng-untouched ng-pristine ng-valid"
          >
            <option
              value="12 hours"
            >
               12 hours 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="1 day"
            >
               1 day 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="3 days"
            >
               3 days 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="1 week"
            >
               1 week 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="Customize"
            >
               Customize 
            </option>
          </select>
        </div>
      </div>
    </div>
    <br />
  </div><div
    class="modal-footer"
  >
    <button
      class="btn btn-light"
      type="button"
    >
      No, cancel the Operation
    </button>
    <button
      class="btn btn-success modal-btn-ok"
      disabled=""
      type="button"
    >
       Save Changes
    </button>
  </div>
</tm-individual-extension-date-modal>
`;

exports[`IndividualExtensionDateModalComponent should snap with the extended students 1`] = `
<tm-individual-extension-date-modal
  DATETIME_FORMAT={[Function String]}
  ONE_DAY_IN_MILLISECONDS={[Function Number]}
  ONE_HOUR_IN_MILLISECONDS={[Function Number]}
  ONE_MINUTE_IN_MILLISECONDS={[Function Number]}
  RadioOptions={[Function Object]}
  activeModal={[Function NgbActiveModal]}
  datePicker={[Function Object]}
  extendByDatePicker={[Function Object]}
  extendByDeadlineKey=""
  extendByDeadlineOptions={[Function Map]}
  feedbackSessionEndingTime="0"
  feedbackSessionTimeZone=""
  numInstructors={[Function Number]}
  numStudents={[Function Number]}
  onConfirmCallBack={[Function EventEmitter_]}
  radioOption={[Function Number]}
  simpleModalService={[Function SimpleModalService]}
  sortMapByOriginalOrder={[Function Function]}
  timePicker={[Function Object]}
  timeZoneService={[Function TimezoneService]}
>
  <div
    class="modal-header alert-primary"
  >
    <h5
      class="modal-title"
    >
      Select the new deadline for extension
    </h5>
    <button
      class="close"
      type="button"
    >
      <i
        class="fas fa-times"
      />
    </button>
  </div><div
    class="modal-body"
  >
     Extending the feedback session for 
    <b>
      10
    </b>
     student(s) and 
    <b>
      20
    </b>
     instructor(s):
  </div><div
    class="container ml-1"
  >
    <div
      class="row pl-3"
    >
      <label
        class="mr-2"
      >
        <input
          class="ng-untouched ng-pristine ng-valid"
          name="optradio"
          type="radio"
        />
         Extend deadline by 
      </label>
      <label>
        <input
          class="ng-untouched ng-pristine ng-valid"
          name="optradio"
          type="radio"
        />
         Extend deadline to 
      </label>
    </div>
    <div>
      <small>
        Extend the feedback session by:
      </small>
      <div
        id="submission-start-date"
      >
        <div
          class="input-group w-75"
        >
          <select
            class="form-control ng-untouched ng-pristine ng-valid"
          >
            <option
              value="12 hours"
            >
               12 hours 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="1 day"
            >
               1 day 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="3 days"
            >
               3 days 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="1 week"
            >
               1 week 
              <div>
                 (5 Apr 2000 2:00:00) 
              </div>
            </option>
            <option
              value="Customize"
            >
               Customize 
            </option>
          </select>
        </div>
      </div>
    </div>
    <br />
  </div><div
    class="modal-footer"
  >
    <button
      class="btn btn-light"
      type="button"
    >
      No, cancel the Operation
    </button>
    <button
      class="btn btn-success modal-btn-ok"
      disabled=""
      type="button"
    >
       Save Changes
    </button>
  </div>
</tm-individual-extension-date-modal>
`;
